### PR TITLE
Fix build with GCC 13 -Werror

### PR DESCRIPTION
**Pull request by mcatanzaro**
#### 2b34340acfdb25e9451fa6885ab24727ac1718b8
<pre>
Fix build with GCC 13 -Werror
<a href="https://bugs.webkit.org/show_bug.cgi?id=254758">https://bugs.webkit.org/show_bug.cgi?id=254758</a>

Reviewed by Adrian Perez de Castro.

This allows building with GCC 13 in DEVELOPER_MODE (where warnings are
now fatal).

* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp:
(JSC::BytecodeGenerator::BytecodeGenerator):
* Source/JavaScriptCore/heap/AbstractSlotVisitorInlines.h:
(JSC::AbstractSlotVisitor::ReferrerContext::ReferrerContext):
* Source/WebCore/animation/DocumentTimeline.cpp:
(WebCore::DocumentTimeline::animationCanBeRemoved):
* Source/WebCore/platform/animation/TimingFunction.h:
* Source/WebCore/platform/graphics/AudioTrackPrivate.h:
(WebCore::AudioTrackPrivate::operator== const):
* Source/WebCore/platform/graphics/InbandTextTrackPrivate.h:
(WebCore::InbandTextTrackPrivate::operator== const):
* Source/WebCore/platform/graphics/TrackPrivateBase.h:
* Source/WebCore/platform/graphics/VideoTrackPrivate.h:
(WebCore::VideoTrackPrivate::operator== const):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::leftToRightDeclaration):
(WebCore::Style::rightToLeftDeclaration):
* Source/WebCore/svg/SVGPathSeg.h:
* Source/bmalloc/libpas/src/libpas/pas_fast_megapage_table.c:
(pas_fast_megapage_table_set_by_index):
* Tools/TestWebKitAPI/Tests/WTF/BoxPtr.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WTF/CompactRefPtr.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WTF/NakedPtr.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WTF/PackedRefPtr.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WTF/RefPtr.cpp:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/262664@main">https://commits.webkit.org/262664@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/46a6edfb850b2bad2b86990783ba249ab38bdd96

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2146 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2179 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2246 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/3069 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2192 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2120 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2259 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2224 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1949 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2165 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1920 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1924 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2924 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1905 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1905 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1805 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1790 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1978 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1953 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3095 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2053 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1966 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1766 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/2206 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1892 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1915 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/510 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/543 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2083 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/2256 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/625 "Passed tests") |

### CODE REVIEW

File: Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp [hunk 1]
```diff
@@ -711,7 +711,7 @@ void LineBoxBuilder::adjustOutsideListMarkersPosition(LineBox& lineBox)
     auto rootInlineBoxLogicalLeft = lineBox.logicalRectForRootInlineBox().left();
     auto rootInlineBoxOffsetFromContentBoxOrIntrusiveFloat = lineBoxOffset + rootInlineBoxLogicalLeft;
     for (auto listMarkerBoxIndex : m_outsideListMarkers) {
-        auto& listMarkerRun = lineContent().runs[listMarkerBoxIndex];
+        auto& listMarkerRun = m_lineContent.runs[listMarkerBoxIndex];
```
Comment (aperezdc) [inline Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp]: Is this change really necessary? It does not look like it should be needed.

File: Source/WebCore/platform/animation/TimingFunction.h [hunk 2]
```diff
@@ -138,7 +140,9 @@ class CubicBezierTimingFunction final : public TimingFunction {
 
     static const CubicBezierTimingFunction& defaultTimingFunction()
     {
+IGNORE_GCC_WARNINGS_BEGIN("dangling-reference")
         static const CubicBezierTimingFunction& function = create().leakRef();
```
Comment (aperezdc) [inline Source/WebCore/platform/animation/TimingFunction.h]: Wouldn't it be better to use `NeverDestroyed<>` for these?

**Review by aperezdc — APPROVED** [review]
I left a couple of comments, although nothing that should prevent this from landing.

File: Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp (hunk 1 above)
Comment (mcatanzaro) [reply Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp]: At the time I posted this PR, it was needed to dodge a false-positive -Wdangling-reference warning. But it's not needed anymore after https://commits.webkit.org/262366@main, so I'll remove it before landing.

File: Source/WebCore/platform/animation/TimingFunction.h (hunk 2 above)
Comment (mcatanzaro) [reply Source/WebCore/platform/animation/TimingFunction.h]: Good idea. I'll try that.

Comment (mcatanzaro) [reply Source/WebCore/platform/animation/TimingFunction.h]: `NeverDestroyed<Ref<LinearTimingFunction>>` works.

**Comment by aperezdc** [comment]
@mcatanzaro The style checker is complaining, but otherwise good to go. Nice that the change to `NeverDestroyed<>` works well 👍🏼

**Comment by mcatanzaro** [comment]
Odd, I thought I had added the merge-queue label last night... maybe there was a race condition.

> The style checker is complaining, but otherwise good to go. Nice that the change to `NeverDestroyed<>` works well 👍🏼

I don't want to reformat the libpas code.
